### PR TITLE
Adding rest-like web api and changing to Json.NET

### PR DESCRIPTION
**Pull request by Maddimax**
This adds get / set value to the http server and two example scripts on how to use the WMI and RestLike API from python

### CODE REVIEW

**Comment by Phyxion** [comment]
Thank you, certainly good idea, I'll add some comments.

File: Utilities/HttpServer.cs [hunk 1]
```diff
@@ -98,6 +101,58 @@ public class HttpServer {
...
 
+    public static IDictionary<string, string> ToDictionary(NameValueCollection col) {
+      IDictionary<string, string> dict = new Dictionary<string, string>();
+      foreach (var k in col.AllKeys) {
+        dict.Add(k, col[k]);
+      }
+      return dict;
+    }
+
+    public SensorNode findSensor(Node node, string id) {
```
Comment (Phyxion) [inline Utilities/HttpServer.cs]: Can you change this to C# formatting (i.e., FindSensor)?

File: Utilities/HttpServer.cs [hunk 2]
```diff
@@ -98,6 +101,58 @@ public class HttpServer {
...
+      foreach (Node child in node.Nodes) {
+        SensorNode s = findSensor(child, id);
+        if (s != null) {
+          return s;
+        }
+      }
+      return null;
+    }
+
+    public bool doSetValue(string id, string value) {
```
Comment (Phyxion) [inline Utilities/HttpServer.cs]: Maybe something like 'SetSensorControlValue'?

File: Utilities/HttpServer.cs [hunk 3]
```diff
@@ -98,6 +101,58 @@ public class HttpServer {
...
+        else {
+          sNode.Sensor.Control.SetSoftware(float.Parse(value));
+        }
+        return true;
+      }
+
+      return false;
+    }
+
+    public string getValue(string id) {
```
Comment (Phyxion) [inline Utilities/HttpServer.cs]: GetSensorValue?

File: Utilities/HttpServer.cs [hunk 4]
```diff
@@ -113,6 +168,38 @@ public class HttpServer {
 
       HttpListenerRequest request = context.Request;
 
+      if (request.HttpMethod == "POST") {
```
Comment (Phyxion) [inline Utilities/HttpServer.cs]: Going to just put a comment for the entire block here, but can you add some comments to this block so we know what is going on? I get the general idea here but some example output would be welcome.

File: Utilities/HttpServer.cs [hunk 5]
```diff
@@ -246,6 +333,7 @@ public class HttpServer {
       JSON += "]";
 
       if (n is SensorNode) {
+        JSON += ", \"id\": \"" + ((SensorNode)n).Sensor.Identifier.ToString() + "\"";
```
Comment (Phyxion) [inline Utilities/HttpServer.cs]: I think we should follow the other output here and use "Id".

**Comment by Maddimax** [comment]
Hey, I've cleaned up and improved to code, and added some description into the commit message.

File: Utilities/HttpServer.cs (hunk 4 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: Are you ok with the commit message, or should I add comments to the code?

File: Utilities/HttpServer.cs (hunk 5 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: Agreed, fixed.

File: Utilities/HttpServer.cs (hunk 1 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: Done.

File: Utilities/HttpServer.cs (hunk 2 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: Renamed to SetSensorControlValue.

File: Utilities/HttpServer.cs (hunk 3 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: Renamed to GetSensorValue.

File: Utilities/HttpServer.cs (hunk 4 above)
Comment (Phyxion) [reply Utilities/HttpServer.cs]: Comments in the code is better as it's not that easy to read back the comments from the commit. I think some general comments regarding the input and output should be the minimum 👍

Comment (Maddimax) [reply Utilities/HttpServer.cs]: Looking at the previous code, there are basically no comments (so I would like to stay with that style). And I'd rather have the code speak for itself in regards to its functionality. If you think there is something unclear I'd rather change the code to make it more obvious what is happening ( which also was the reason to split the block into multiple functions )

File: Utilities/HttpServer.cs [hunk 6]
```diff
@@ -98,6 +102,118 @@ public class HttpServer {
...
+
+      try {
+        if(request.Url.Segments.Length > 1) {
+          if (request.Url.Segments[1] == "Sensor") {
+            result = HandleSensorRequest(request);
+          }
+        }
+      }
+      catch {
+        result = "{ \"result\" : \"Internal Error\" }";
```
Comment (Maddimax) [inline Utilities/HttpServer.cs]: TODO: Fix this to follow the { result / message } style

**Comment by Maddimax** [comment]
I've added the package "Json.NET". Sadly I had to commit its file for the build check to succeed. Is there a way to change it so the builder downloads the NuGet package itself ?

File: Utilities/HttpServer.cs (hunk 4 above)
Comment (Maddimax) [reply Utilities/HttpServer.cs]: I've added comments into the code now.

**Comment by Phyxion** [comment]
Looks good now, thank you.

**Pull request merged**